### PR TITLE
Some fixes for *BSD systems.

### PR DESCRIPTION
**Pull request by mazocomp**
Depend on find_package() function, do not manually pass -I and -L flags to both compiler and linker.
Allow to build with both D-Bus and filteraudio.
Allow to build and run tests.
Polish build documentation, mention NetBSD and DragonFlyBSD.
Do not set CMAKE_OSX_DEPLOYMENT_TARGET on non-OSX systems.
Fix link-time error at FreeBSD.
Ignore *.orig files, they are generated by patch(1) utility as backup files.
Remove unused OPENBSD variable from CMakeLists.txt.
Do not print "Enable DBus" and "Enable Tests" at Windows and do not print "WIndows Legacy" at not Windows systems.
Enable tests by default.
Fix DragonFlyBSD build.
Make uTox more portable!

---
This change is [Reviewable](https://reviewable.io/reviews/utox/utox/1281)

### CODE REVIEW

File: docs/BUILD.md [hunk 1]
```diff
@@ -127,36 +121,31 @@ cd uTox/
...
+
 Now compile uTox:
 
 ```bash
 git clone --recursive git://github.com/uTox/uTox.git
 cd uTox/
 mkdir build
 cd build
-cmake ..
+env CC=gcc cmake ..
```
Comment (GrayHatter) [inline docs/BUILD.md]: I don't want to specify that you need to use gcc here. It should work with any compiler. I know per the IRC conversation that it worked with GCC, but unless you can tell me WHY it only works with GCC. I don't want this in the docs.

File: CMakeLists.txt [hunk 2]
```diff
@@ -303,6 +301,9 @@ elseif(APPLE)
     add_definitions("-x objective-c")
     add_subdirectory(src/cocoa)
 elseif(UNIX)
+    find_package(X11 REQUIRED)
```
Comment (GrayHatter) [inline CMakeLists.txt]: why is this here?

Comment (mazocomp) [reply CMakeLists.txt]: @GrayHatter Because otherwise I have to write the same thing into src/ui/CMakeLists.txt, src/av/CMakeLists.txt, src/layout/CMakeLists.txt and so on in order to pass -I/usr/X11R6/include to compiler in OpenBSD and -I/usr/X11R7/include to compiler in NetBSD, think of this as a local and global scopes, cmake is not that bad as you describe.

File: docs/BUILD.md (hunk 1 above)
Comment (mazocomp) [reply docs/BUILD.md]: It builds with Clang too, but: https://github.com/uTox/uTox/issues/1234

**Comment by mazocomp** [comment]
I am super porter!
